src/ontology/mondo-ingest.Makefile

```diff
@@ -498,6 +499,41 @@ slurp-all-no-updates: $(foreach n,$(ALL_COMPONENT_IDS), slurp-no-updates-$(n))
 .PHONY: slurp-all
 slurp-all: $(foreach n,$(ALL_COMPONENT_IDS), slurp-$(n))
 
+
+#############################
+###### Synchronization ######
```

Comment:
## General comments
@matentzn Just responding to your [general / review closing comment](https://github.com/monarch-initiative/mondo-ingest/pull/363/#pullrequestreview-1792756509) as a thread instead so that it can be marked resolved

### Docstrings and Comments
<details><summary>Details</summary>
<p>

The only long one is in `_convert_edge_namespace()`:
```
    failed_lookups: Each entry in this set can be 1 of 2 cases: (a) the edge exists in one ontology, represented by the
    namespace of IDs in rels & the keys in ns_map, but it does not exist in the mapped otology, represented by the
    namespace of IDs in rels2 & the values in ns_map; it may simply be that there are some terms in the 1st ontology
    that do not exist in the 2nd ontology, or (b) the edges should be in both ontologies, but if the inputs
    to this pipeline are out of sync, the mapped terms did not appear in ns_map.
```

Probably could be a bit more concise. But generally, I am fond of detailed docstrings. If you look at popular libraries, they often have very long docstrings. I don't mind them because they can be collapsed more easily, and there is really nowhere better to put them. You could put in external docs, but that's a hassle because it's not linked to the code, and it also just pollutes the docs with stuff that's not going to be relevant to most people. I think erring on the side of verbose is good. Generally people only read these on the rare occasion where they're digging deeply for something, and when it comes to digging deeply, you want to have access to as much information as possible.

</p>
</details> 

### `todo` comments
<details><summary>Details</summary>
<p>

I have a convention where I distinguish between `TODO` and `todo`. I use `TODO` for something that is important; e.g. it is necessary for completing the development objective, or it is related to a bug. I use `todo` for things like optimizations, refactoring, additional testing, and ideas.

Everything that's left here is a `todo`, so not a major worry.

Also I should note that I differ from other developers in that I am in favor of leaving (lots of) `todo`s even in production. For these low priority things, there are basically 3 choices:
a. Don't leave anything written about them
b. Put in a GitHub issue
c. Leave in the code
d. Leave written elsewhere

For 'b'. Basically, I don't think everything is important enough to go into a GitHub issue, though perhaps these kind of things could go into 1+ issues marked such as "low priority misc".

I think 'd' is a bad idea.

I think 'a' is not a good idea. Similar to my advocation of long docstrings, I think it is better when more is said than not.

So I advocate for 'c'.


</p>
</details> 

## Global variables
<details><summary>Details</summary>
<p>
Not confident on the best solutions but I offer my thoughts here.

Mostly all of the global variables are used only within the global dictionary `EX_DEFAULTS`. The design decision for this being a global variable, even though it's really only used in `cli()`, is because it is essentially a kind of configuration, configs are typically placed at the top of a file  by convention for easy access.

I would advocate moving to a separate file in this case because this config is specific to the sync subclass script and no other files.

I suppose it is actually not great that you can't tell just looking at the code whether or not the globals are re-used in a lot of places. I am not a fan of global variables, myself.

A potential solution is to create a `config()` or `defaults()` function and place everything in there. I don't like it aesthetically, but perhaps it is better.

I'd advocate against environmental variables because this makes collaboration more difficult. Even for me, I program on multiple machines sometime, and keeping these files in sync would add additional hassel for me.

Also there can probably be a little cleanup here. Some of these vars could be removed.

</p>
</details> 

### Function length
<details><summary>Details</summary>
<p>

Yeah I think `sync_subclassof()` is really the only long one. Actually I was thinking it needed to be broken down too. It's actually only about 200 lines, and about 50% of that is comments, so it is not actually that long. But it is pretty complex.

I've been thinking about making it smaller, and this was the idea I had. Maybe you agree?: Each of the "cases" (about 20 lines each) gets its own function.

Beyond that, I think it would be difficult to make this much smaller, though I do have a `todo` related to doing an OO refactor that would simplify things a bit / reduce lines here, but it would not be a quick change.

</p>
</details> 

### Hardcoded Values
<details><summary>Details</summary>
<p>

I put `EX_` to stand for "example". The concerns you bring up here, I have addressed in THIS COMMENT.

</p>
</details> 

### Unused Code
<details><summary>Details</summary>
<p>

I generally agree with this, and I have a `todo` about this at the top of the block of unused code. Much of this code would become relevant should we do some additional cases. But I can stash this code in my notes somewhere.

</p>
</details>

Reply:
Thank you for your thoughtful comments! I wanted you to be aware of my thoughts, and trust you with the execution! I am quite liberal, maybe to the border of anarchical, when it comes to code, and @twhetzel may be implementing stricter code standards moving forward. This was the last PR where I set the tone of the review, so I am happy if you are.